mailpoet/lib/Util/Notices/DisabledMailFunctionNotice.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace MailPoet\Util\Notices;
+
+use MailPoet\Mailer\Mailer;
+use MailPoet\Settings\SettingsController;
+use MailPoet\Util\Helpers;
+use MailPoet\Util\License\Features\Subscribers as SubscribersFeature;
+use MailPoet\WP\Functions as WPFunctions;
+use MailPoet\WP\Notice;
+
+class DisabledMailFunctionNotice {
+
+  const OPTION_NAME = 'disabled-mail-function';
+
+  /** @var SettingsController */
+  private $settings;
+
+  /** @var WPFunctions */
+  private $wp;
+
+  /** @var SubscribersFeature */
+  private $subscribersFeature;
+
+  public function __construct(
+    WPFunctions $wp,
+    SettingsController $settings,
+    SubscribersFeature $subscribersFeature
+  ) {
+    $this->settings = $settings;
+    $this->wp = $wp;
+    $this->subscribersFeature = $subscribersFeature;
+  }
+
+  public function init($shouldDisplay): ?string {
+    $shouldDisplay = $shouldDisplay && $this->checkRequirements();
+    if (!$shouldDisplay) {
+      return null;
+    }
+
+    return $this->display();
```

Comment:
Just a nitpick but I noticed that we are returning something in this method but we don't do anything with what is returned when init() is called. Do we need to return something?

Reply:
We are not using the returned data in the class.
It is being used [here](https://github.com/mailpoet/mailpoet/blob/b63e19b7d260957013a37a501de2e709e2cb480c/mailpoet/tests/integration/Util/Notices/DisabledMailFunctionNoticeTest.php#L54-L61) for testing. i.e confirming we get the expected notice message

Reply:
Ah, I missed that it is being used for testing. I'm not a huge fan of class behavior that is only used in the tests, but I see that this was already used in the other notice classes and I don't think we need to address this here. I will move this PR to QA.